nitro-node/src/nitro.ts

```diff
@@ -0,0 +1,511 @@
+  // Retry the request up to 3 times if it fails, with a delay of 500 milliseconds between retries.
+  const response = await fetchRetry(NITRO_HTTP_VALIDATE_MODEL_URL, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    retries: 5,
+    retryDelay: 500,
+  });
+  log(
+    `[NITRO]::Debug: Validate model state success with response ${JSON.stringify(
+      response,
+    )}`,
+  );
+  // If the response is OK, check model_loaded status.
+  if (response.ok) {
+    const body = await response.json();
+    // If the model is loaded, return an empty object.
+    // Otherwise, return an object with an error message.
+    if (body.model_loaded) {
+      return {};
+    }
+  }
+  return { error: "Validate model status failed" };
+}
+
+/**
+ * Terminates the Nitro subprocess.
+ * @returns A Promise that resolves when the subprocess is terminated successfully, or rejects with an error message if the subprocess fails to terminate.
+ */
+export async function killSubprocess(): Promise<NitroModelOperationResponse> {
+  const controller = new AbortController();
+  setTimeout(() => controller.abort(), 5000);
+  log(`[NITRO]::Debug: Request to kill Nitro`);
+
+  try {
+    const _response = await fetch(NITRO_HTTP_KILL_URL, {
+      method: "DELETE",
+      signal: controller.signal,
+    });
+    subprocess?.kill();
+    subprocess = undefined;
+    await tcpPortUsed.waitUntilFree(PORT, 300, 5000);
+    log(`[NITRO]::Debug: Nitro process is terminated`);
+    return {};
+  } catch (err) {
+    return { error: err };
+  }
+}
+
+/**
+ * Spawns a Nitro subprocess.
+ * @returns A promise that resolves when the Nitro subprocess is started.
+ */
+export function spawnNitroProcess(): Promise<NitroModelOperationResponse> {
+  log(`[NITRO]::Debug: Spawning Nitro subprocess...`);
+
+  return new Promise(async (resolve, reject) => {
+    const executableOptions = executableNitroFile(nvidiaConfig, binPath);
+
+    const args: string[] = ["1", LOCAL_HOST, PORT.toString()];
+    // Execute the binary
+    log(
+      `[NITRO]::Debug: Spawn nitro at path: ${executableOptions.executablePath}, and args: ${args}`,
+    );
+    subprocess = spawn(
+      executableOptions.executablePath,
+      ["1", LOCAL_HOST, PORT.toString()],
+      {
+        cwd: binPath,
+        env: {
+          ...process.env,
+          CUDA_VISIBLE_DEVICES: executableOptions.cudaVisibleDevices,
+        },
+      },
+    );
+
+    // Handle subprocess output
+    subprocess.stdout.on("data", (data: any) => {
+      log(`[NITRO]::Debug: ${data}`);
+    });
+
+    subprocess.stderr.on("data", (data: any) => {
+      log(`[NITRO]::Error: ${data}`);
+    });
+
+    subprocess.on("close", (code: any) => {
+      log(`[NITRO]::Debug: Nitro exited with code: ${code}`);
+      subprocess = undefined;
+      reject(`child process exited with code ${code}`);
+    });
+
+    tcpPortUsed.waitUntilUsed(PORT, 300, 30000).then(() => {
```

Comment:
Sorry for the confusion, I don't think spawning a nitro subprocess should take that long. Please help adjust.